### PR TITLE
Bug 1689610 - replace passFailRatio with totalFailures/totalJobs

### PR DESCRIPTION
**Pull request by jmaher**

### CODE REVIEW

File: treeherder/push_health/tests.py [hunk 1]
```diff
@@ -146,13 +139,19 @@ def get_current_test_failures(push, option_map, jobs, investigatedTests=None):
                 'suggestedClassification': 'New Failure',
                 'confidence': 0,
                 'tier': job.tier,
+                'totalFailures': 0,
+                'totalJobs': 0,
```
Comment (bhearsum) [inline treeherder/push_health/tests.py:143]: I'm a bit surprised that a field called `totalJobs` ends up excluding results other than `success` and `testfailed`. I'm assuming that this means, eg: jobs that had an exception (like malformed payload) wouldn't be counted here). Is there a better name we could use for this?

Maybe I'm misunderstanding that possible states here, though.

File: treeherder/push_health/tests.py [hunk 2]
```diff
@@ -146,13 +139,19 @@ def get_current_test_failures(push, option_map, jobs, investigatedTests=None):
...
                 'failedInParent': False,
-                'passFailRatio': passFailRatio,
```
Comment (bhearsum) [inline treeherder/push_health/tests.py:150]: What consumers do we have of this data currently? Is it safe to remove it right away? It might be better to keep the old code in, and only remove it once the consumers have updated.

**Review by bhearsum — COMMENTED** [review]
I think @sarah-clements should review this too; I'm not comfortable enough r+ing a change like this on my own.

File: treeherder/push_health/tests.py (hunk 1 above)
Comment (jmaher) [reply treeherder/push_health/tests.py:143]: cancelled, pending, etc.  the main goal is to count total jobs that ran the test- this is an edge case and makes an assumption that if a job ran 3 times that a given test in that job ran 3 times as well.

File: treeherder/push_health/tests.py (hunk 2 above)
Comment (jmaher) [reply treeherder/push_health/tests.py:150]: this might make sense; my understanding is that only pushhealth consumes this, but with an API you never know.  I have added Sarah to the reviewer list and will see what else shakes out.

File: treeherder/push_health/tests.py (hunk 1 above)
Comment (bhearsum) [reply treeherder/push_health/tests.py:143]: Ah, I see! So this is a count of individual tests running, not tasks. In that case, I think the naming makes sense since cancelled tasks, etc. don't actually run the tests.

Comment (jmaher) [reply treeherder/push_health/tests.py:143]: maybe I misrepresented it:
totalFailures = total times this test case failed
totalJobs = total completed jobs which the test belongs to

Comment (bhearsum) [reply treeherder/push_health/tests.py:143]: Right, sorry. That's what I meant.

File: treeherder/push_health/tests.py (hunk 2 above)
Comment (sarah-clements) [reply treeherder/push_health/tests.py:150]: I looked in papertrail and I didn't see any GET requests for the `health` and `health_summary` APIs that didn't originate from Treeherder so this is probably an ok change to make. We do only have 3 days of active data in the logs before its archived though.

**Review by sarah-clements — APPROVED** [review]
Thanks for the detailed explanation in the bug. This change is probably fine but interestingly, it looks like in Cam's [mozci pr](https://github.com/mozilla/treeherder/pull/6763/files#diff-3651aec6dcf76b83e1e031e03f93d03dbb1861cc63186fc1bc73dea5ced9c9fcL112) the calculation in question and `passFailRatio` field was changed. However, I still see the error mentioned in your bug bucketed into "needs investigation" in his branch on [prototype](https://treeherder-prototype.herokuapp.com/push-health/push?revision=401b48258d9e24c29afa908506c909ec2f4b5a67&repo=try&testGroup=pr&selectedTest=toolkitcomponentsextensionstestmochitesttestchromeextcontentscripttelemetryhtml&selectedTaskId=&selectedJobName=toolkit%2Fcomponents%2Fextensions%2Ftest%2Fmochitest%2Ftest_chrome_ext_contentscript_telemetry.html+test-macosx1014-64-qr%2Fopt-mochitest-chrome-1proc-2) (currently live). So I'm curious what you make of that. 

Btw, status of that mozci pr is the package needs a bump and I need to check if there are any recurring timeout issues (hope to get to that tomorrow). But it also wouldn't hurt to have another set of eyes looking at that code.

**Comment by jmaher** [comment]
I took a look at prototype vs production, things are pretty much identical, including the passfailratio and issues when a task fails for another reason that it ends up being counted against the total failures for the specific issue.  A clear example is:
https://treeherder.mozilla.org/jobs?repo=try&selectedTaskRun=E3ukfbbpQyCDVMl1-Th_ww.1&tier=1%2C2%2C3&revision=04999ae2f8e615798fbd05c785830988dd1e3a67&searchStr=bc1

and on prototype:
https://treeherder-prototype.herokuapp.com/push-health/push?revision=04999ae2f8e615798fbd05c785830988dd1e3a67&repo=try&testGroup=ki&selectedTest=dommediaautoplaytestmochitesttestautoplaypolicyhtml&selectedTaskId=

Looking in detail at the api returned values the passfailratio is either 0.0 (test failures), or >=1.0 for known failures

here 3 of the 5 bc1 jobs fail, but 1 is a generic timeout and the other 2 are specific tests- but the other failure (a crash) is misleading in the push health UI.  

Given this, I don't see the mozci changes affecting the way we summarize total failures seen and total jobs completed, it most likely will help classify things more accurately and at least in a similar way to what the sheriffs see.

Let me know if I should merge this, or make any changes.

**Comment by sarah-clements** [comment]
> I took a look at prototype vs production, things are pretty much identical, including the passfailratio and issues when a task fails for another reason that it ends up being counted against the total failures for the specific issue. A clear example is:
> https://treeherder.mozilla.org/jobs?repo=try&selectedTaskRun=E3ukfbbpQyCDVMl1-Th_ww.1&tier=1%2C2%2C3&revision=04999ae2f8e615798fbd05c785830988dd1e3a67&searchStr=bc1
> 
> and on prototype:
> https://treeherder-prototype.herokuapp.com/push-health/push?revision=04999ae2f8e615798fbd05c785830988dd1e3a67&repo=try&testGroup=ki&selectedTest=dommediaautoplaytestmochitesttestautoplaypolicyhtml&selectedTaskId=
> 
> Looking in detail at the api returned values the passfailratio is either 0.0 (test failures), or >=1.0 for known failures
> 
> here 3 of the 5 bc1 jobs fail, but 1 is a generic timeout and the other 2 are specific tests- but the other failure (a crash) is misleading in the push health UI.
> 
> Given this, I don't see the mozci changes affecting the way we summarize total failures seen and total jobs completed, it most likely will help classify things more accurately and at least in a similar way to what the sheriffs see.
> 
> Let me know if I should merge this, or make any changes.

Ok, this looks good to merge.
